### DISCUSSION TITLE
AKKA.Hosting - how to activate cluster listener

### DISCUSSION THREAD

**Discussion by JTrotta** [akkadotnet/akka.net#6333] [Q&A] [answered]
How to activate cluster listener actor, before cluster is formed using Akka.Hosting .WithClustering extension?

**Comment by Arkatufus** [comment, marked as answer]
When you say cluster listener actor, do you mean using `Cluster.RegisterOnMemberUp()` or having an actor that listens to cluster events?
In both case, it is safe to start the actor inside the `.WithActors()` extension method.

**Comment by JTrotta** [comment]
I mean an actor listening to cluster events,
That's what I did, but no events fired:
`            configurationBuilder
                .WithRemoting(configutationSettings.HostName, configutationSettings.ClusterSettings.Port)
                .WithClustering(new ClusterOptions()
                {
                    Roles = configutationSettings.ClusterSettings.Roles,                    
                    SeedNodes = configutationSettings.ClusterSettings?.Seeds,
                })
                .WithDistributedPubSub(configutationSettings.ClusterSettings.Roles[0])
                .WithActors((system, registry, resolver) =>
                {
                    var clusterListener = system.ActorOf(Props.Create(() => new ClusterListener()), "clusterlistener");
                    registry.TryRegister<ClusterListener>(clusterListener);
                });`

**Comment by netclectic** [comment]
What is the difference between `WithActors` and `StartActors`?
I'm starting my cluster listener in StartActors and I've not had any issues.

**Reply by Arkatufus** [reply]
They are practically the same

**Comment by JTrotta** [comment]
Sorry, it was a loggger configuration error.
It works as expected
thank you both.
